src/sentry/api/endpoints/organization_spans_aggregation.py

```diff
@@ -0,0 +1,230 @@
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        transaction = request.query_params.get("transaction", None)
+        if transaction is None:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST, data={"details": "Transaction not provided"}
+            )
+
+        builder = QueryBuilder(
+            dataset=Dataset.SpansIndexed,
+            params=params,
+            selected_columns=["transaction_id", "count()"],
+            query=f"transaction:{transaction}",
+            limit=100,
+        )
+        builder.columns.append(
+            Function(
+                "groupArray",
+                parameters=[
+                    Function(
+                        "tuple",
+                        parameters=[
+                            Column("span_id"),
+                            Column("is_segment"),
+                            Column("parent_span_id"),
+                            Column("group"),
+                            Column("group_raw"),
+                            Column("description"),
+                            Column("start_ms"),
+                            Column("duration"),
+                            Column("exclusive_time"),
+                        ],
+                    )
+                ],
+                alias="spans",
+            )
+        )
+        snql_query = builder.get_snql_query()
+        snql_query.tenant_ids = {"organization_id": organization.id}
+        results = raw_snql_query(snql_query, Referrer.API_ORGANIZATION_SPANS_AGGREGATION.value)
+        for event in results["data"]:
+            span_tree = {}
+            root_span_id = None
+            spans = event["spans"]
+
+            for span in spans:
+                span = EventSpan(*span)
+                span_id = getattr(span, "span_id")
+                is_root = getattr(span, "is_segment")
+                if is_root:
+                    root_span_id = span_id
+                if span_id not in span_tree:
+                    spans_dict = span._asdict()
+                    # Fallback to group_raw if group doesn't exist for now
+                    spans_dict["coalesced_group"] = (
+                        spans_dict["group_raw"]
+                        if spans_dict["group"] == NULL_GROUP
+                        else spans_dict["group"]
+                    )
+                    span_tree[span_id] = spans_dict
+                    span_tree[span_id]["children"] = []
+
+            for span in span_tree.values():
+                parent_id = span["parent_span_id"]
+                try:
+                    parent_span = span_tree[parent_id]
+                    children = parent_span["children"]
+                    children.append(span)
+                except KeyError:
```

Comment:
I'm personally not a fan of trying of using try/excepts to contain logic. One of the reasons being i'm not sure if you're trying to check either `parent_id in span_tree` or `"children" in parent_span`. And the other being me being a purist and try/except should be for errors :P

Reply:
https://github.com/getsentry/sentry/pull/56291/commits/1fbbfe05516aba2dd890cb846b44a7af04a3de77#diff-1df1994c8fddccf95629e8b491f1ff17c43534b3634dd396256fc448124d7cddR123